src/app/modules/submission/submission.service.ts

```diff
@@ -251,17 +258,64 @@ export const copyPendingSubmissionToSubmissions = (
       return okAsync(submission)
     })
     .andThen((pendingSubmission) => {
-      const submission = new SubmissionModel(
-        omit(pendingSubmission, ['_id', 'created', 'lastModified']),
-      )
+      const submissionContent = omit(pendingSubmission, [
+        '_id',
+        'created',
+        'lastModified',
+      ])
+      const submission = new SubmissionModel({
+        // Explicitly copy over the pending submission's _id
+        _id: pendingSubmissionId,
+        ...submissionContent,
+      })
 
       return ResultAsync.fromPromise(submission.save({ session }), (error) => {
         logger.error({
           message: 'Database save submission error',
           meta: logMeta,
           error,
         })
-        return new DatabaseError(getMongoErrorMessage(error))
+
+        const databaseError = transformMongoError(error)
+
+        return databaseError instanceof DatabaseDuplicateKeyError
+          ? databaseError
+          : new DatabaseError(getMongoErrorMessage(error))
+      }).orElse((error) => {
+        if (error instanceof DatabaseDuplicateKeyError) {
```

Comment:
seems to me that the first `instanceof DatabaseDuplicateKeyError` check is redundant?

Reply:
`error` could be either `DatabaseDuplicateKeyError` or `DatabaseError`. Only in the former case do we want to retry, in the latter case we don't. Am i misunderstanding what you're referring to?

Reply:
Oh, as in because the `DatabaseError` can be wrapped in line 318 instead, right? And right after the `orElse`, the case of `if (error instanceof DatabaseDuplicateKeyError)` is handled, so the return statement here can just `return databaseError`?

Reply:
oh, i see what you mean... hmm, either is ok i think, my view was that it helps to clarify the types of the errors going into the `orElse` more explicitly (to be sure that it's one of `DatabaseDuplicateKeyError` or `DatabaseError`). But you're right that we could be less verbose and achieve the same effect!

Reply:
[ea85744](https://github.com/opengovsg/FormSG/pull/6088/commits/ea857445b34cc99eb35b3e00e3c16f278f91ed4b)